ERCS/erc-7649.md

```diff
@@ -0,0 +1,410 @@
+	 *
+	 * @dev Depending on the implementation, ERC-20 token payment may be required instead of ETH.
+	 *      In such a case, implementation must through if ETH is sent, effectively overriding
+	 *      the function definition as non-payable
+	 *
+	 * @param amount amount of the shares to buy
+	 */
+	function buyShares(uint256 amount) external payable;
+
+	/**
+	 * @notice Buy `amount` of shares in the favor of the address specified (beneficiary).
+	 *      Sender has to supply `getBuyPriceAfterFee(amount)` ETH.
+	 *      First share can be bought only by current subject issuer.
+	 *
+	 * @dev Depending on the implementation, ERC-20 token payment may be required instead of ETH.
+	 *      In such a case, implementation must through if ETH is sent, effectively overriding
+	 *      the function definition as non-payable
+	 *
+	 * @param amount amount of the shares to buy
+	 * @param beneficiary an address receiving the shares
+	 */
+	function buySharesTo(uint256 amount, address beneficiary) external payable;
+
+	/**
+	 * @notice Sell `amount` of shares. Sender gets `getSellPriceAfterFee(amount)` of ETH.
+	 *      Last share cannot be sold.
+	 *
+	 * @dev Depending on the implementation, ERC-20 token may be payed instead of ETH.
+	 *
+	 * @param amount amount of the shares to sell
+	 */
+	function sellShares(uint256 amount) external;
+
+	/**
+	 * @notice Sell `amount` of shares in the favor of the address specified (beneficiary).
+	 *      The beneficiary gets `getSellPriceAfterFee(amount)` of ETH.
+	 *      Last share cannot be sold.
+	 *
+	 * @dev Depending on the implementation, ERC-20 token may be payed instead of ETH.
+	 *
+	 * @param amount amount of the shares to sell
+	 * @param beneficiary an address receiving the funds from the sale
+	 */
+	function sellSharesTo(uint256 amount, address payable beneficiary) external;
+
+	/**
+	 * @notice Cumulative value of all trades; allows to derive cumulative fees paid
+	 *
+	 * @dev This value cannot decrease over time; it can increase or remain constant
+	 *      if no trades are happening
+	 *
+	 * @return Sum of the modulo of all trading operations
+	 */
+	function getTradeVolume() external view returns(uint256);
+```
+
+
+## Rationale
+
+<!--
+  The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages.
+
+  The current placeholder is acceptable for a draft.
+
+  TODO: Remove this comment before submitting
+-->
+
+The proposed standard focuses on enhancing the ERC-721 standard by introducing a framework for embedding bonding
+curve-based liquidity into NFTs. This approach provides creators with a flexible and customizable tool to attract
+liquidity through bonding curve mechanisms, while ensuring creators receive guaranteed fees for their contributions.
```

Comment:
This belongs in the Motivation section. Rationale should explain the technical choices made in this proposal.